lib/repo/git_repository.rb

```diff
@@ -66,8 +74,59 @@ def self.create(connect_string)
                           some directory with same name exists already")
       end
 
-      #Create it (we're not going to use a bare repository)
-      repo = Rugged::Repository.init_at(connect_string)
+      ga_repo = Gitolite::GitoliteAdmin.new(Repository.conf[:REPOSITORY_STORAGE] + '/gitolite-admin', GITOLITE_SETTINGS)
+
+      # Bring the repo up to date
+      ga_repo.reload!
+
+      # Grab the gitolite admin repo config
+      conf = ga_repo.config
+
+      repo_name = File.basename(connect_string.split('/').last)
+
+      # Grab the repo in question, if it does not exist, create it
+      repo = ga_repo.config.get_repo(repo_name)
+      if repo.nil?
+          # Generate new repo since this repo hasn't been created yet
+          repo = Gitolite::Config::Repo.new(repo_name)
+      end
+
+      # Add permissions for git user
+      repo.add_permission("RW+", "", "git") # Give 'git' user access to all repos
```

Comment:
Line is too long. [81/80]<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.